source/Halibut.Tests/UsageFixture.cs

```diff
@@ -35,37 +35,23 @@ public void OctopusCanDiscoverTentacle()
                 info.RemoteThumbprint.Should().Be(Certificates.TentacleListeningPublicThumbprint);
             }
         }
-       
+
 
         [Test]
-        public void OctopusCanSendMessagesToListeningTentacle()
+        [TestCaseSource(typeof(ServiceConnectionTypesToTest))]
+        public void OctopusCanSendMessagesToTentacle(ServiceConnectionType serviceConnectionType)
         {
-            using (var clientAndServer = ClientServiceBuilder.Listening().WithServiceFactory(GetDelegateServiceFactory()).Build())
+            using (var clientAndServer = ClientServiceBuilder
+                       .ForMode(serviceConnectionType)
+                       .WithServiceFactory(GetDelegateServiceFactory())
+                       .Build())
             {
                 var echo = clientAndServer.CreateClient<IEchoService>();
                 echo.SayHello("Deploy package A").Should().Be("Deploy package A...");
-                var watch = Stopwatch.StartNew();
-                for (var i = 0; i < 2000; i++)
-                {
-                    echo.SayHello("Deploy package A").Should().Be("Deploy package A...");
-                }
 
-                Console.WriteLine("Complete in {0:n0}ms", watch.ElapsedMilliseconds);
-            }
-        }
-
-        [Test]
-        public void OctopusCanSendMessagesToPollingTentacle()
-        {
-            var services = GetDelegateServiceFactory();
-            services.Register<ISupportedServices>(() => new SupportedServices());
-            using (var clientAndService = ClientServiceBuilder.Polling().WithServiceFactory(services).Build())
-            {
-                var svc = clientAndService.CreateClient<ISupportedServices>();
-                for (var i = 1; i < 100; i++)
+                for (var i = 0; i < 2000; i++)
                 {
-                    var i1 = i;
-                    svc.GetLocation(new MapLocation { Latitude = -i, Longitude = i }).Should().Match<MapLocation>(x => x.Latitude == i1 && x.Longitude == -i1);
```

Comment:
Did we stop testing this call? Should we add another test that tests this call?

Reply:
Does it add any value? Listening was doing one thing, Polling another. I made the Listening test do both!! We can add a 
listening and polling variations of this if we want!!

Reply:
I think we should, more tests cost very little.

Reply:
Added the test back in with a polling and listening variation 👍